pkg/internal/testing/addr/manager.go

```diff
@@ -42,16 +42,30 @@ var (
 )
 
 func init() {
-	baseDir, err := os.UserCacheDir()
+	var err error
+	cacheDir, err = tryUserCacheDir()
 	if err != nil {
-		baseDir = os.TempDir()
+		// Either we didn't get a cache directory, or we can't use it
+		cacheDir, err = tryCacheInDir(os.TempDir())
```

Comment:
I would actually vote for trying os.TempDir first, because a) it is more likely to succeed and more importantly b) it gets cleared upon reboot, which is not the case for UserCacheDir in most envs I think?

Reply:
Yup, `UserCacheDir` doesn’t get cleaned up. I wanted to mostly preserve existing behaviour in this PR; were I to change that, I’d actually argue that `UserCacheDir` isn’t really appropriate. It is [defined as](https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html)

> There is a single base directory relative to which user-specific non-essential (cached) data should be written.

(Plus there’s the general idea in XDG that it’s for “real” users in a desktop environment...)

For lock-style files as used here, it seems to me that the appropriate directories are either `/tmp` (as used historically, *e.g.* for X11 locks or SSH agent sockets) or `/run` (but Go doesn’t wrap that).

Reply:
Hm yeah, I am really kinda surprised at seeing this here and wondering why that was done in the first place. Maybe os.Tempdir on MacOS is problematic? I don't have a Mac so I can't test that (quick google didn't show something, but that doesn't mean there is no issue). Do you remember @vincepri ?

Other than that this looks fine, except that the commit and pr title should be prefixed with `:bug:` as documented in the PR template, could you please add that? We also have a check that should verify that, but it appears it is not working.

Reply:
Tempdir on Mac isn't really reliable so we switched to the CacheDir instead